frontend/src/features/admin-form/create/common/CreatePageSidebarContext/CreatePageSidebarContext.tsx

```diff
@@ -0,0 +1,98 @@
+import {
+  createContext,
+  FC,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react'
+
+import {
+  setToInactiveSelector,
+  useBuilderAndDesignStore,
+} from '../../builder-and-design/useBuilderAndDesignStore'
+
+export enum DrawerTabs {
+  Builder,
+  Design,
+  Logic,
+}
+
+type CreatePageSidebarContextProps = {
+  activeTab: DrawerTabs | null
+  isDrawerOpen: boolean
+  handleBuilderClick: () => void
+  handleDesignClick: () => void
+  handleLogicClick: () => void
+  handleClose: () => void
+}
+
+const CreatePageSidebarContext = createContext<
+  CreatePageSidebarContextProps | undefined
+>(undefined)
+
+export const useCreatePageSidebar = (): CreatePageSidebarContextProps => {
+  const context = useContext(CreatePageSidebarContext)
+  if (!context) {
+    throw new Error(
+      `useCreatePageSidebar must be used within a CreatePageSidebarProvider component`,
+    )
+  }
+  return context
+}
+
+export const useCreatePageSidebarContext =
+  (): CreatePageSidebarContextProps => {
+    const [activeTab, setActiveTab] = useState<DrawerTabs | null>(null)
+    const isDrawerOpen = useMemo(
+      () => activeTab !== null && activeTab !== DrawerTabs.Logic,
+      [activeTab],
+    )
+    const setFieldsToInactive = useBuilderAndDesignStore(setToInactiveSelector)
+
+    // Set state to inactive whenever active tab changes
+    useEffect(() => {
+      setFieldsToInactive()
+    }, [activeTab, setFieldsToInactive])
+
+    const handleBuilderClick = useCallback(
+      () => setActiveTab(DrawerTabs.Builder),
+      [setActiveTab],
+    )
+
+    const handleDesignClick = useCallback(
+      () => setActiveTab(DrawerTabs.Design),
+      [setActiveTab],
+    )
+
+    const handleLogicClick = useCallback(
+      () => setActiveTab(DrawerTabs.Logic),
+      [setActiveTab],
+    )
```

Comment:
(react noob question) Is using hooks the best way to achieve this? `setActiveTab` is guaranteed to be stable across invocations, which mean `handleBuilderClick`, `handleDesignClick`, and `handleLogicClick` are also guaranteed to be stable references across invocations, and so in all these `useCallback()` calls, we create local functions which are just discarded. This looks a lot more like construction-time concern rather than render-time concern.

Reply:
Hm, nvm. [This answer](https://reactjs.org/docs/hooks-faq.html#are-hooks-slow-because-of-creating-functions-in-render) indicates that in practice we just shouldn't have to worry about it. It will take me some time to get use to this idea 😅